src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -169,17 +169,15 @@ QgsPostgresConn *QgsPostgresConn::connectDb( const QString &conninfo, bool reado
   // and which is run in a different thread (drag and drop in browser)
   if ( QApplication::instance()->thread() != QThread::currentThread() )
   {
+    // sharing connection between threads is not safe
+    // See https://github.com/qgis/QGIS/issues/21205
     shared = false;
   }
 
   QgsPostgresConn *conn;
 
   if ( shared )
   {
-    // sharing connection between threads is not safe
-    // See https://github.com/qgis/QGIS/issues/21205
-    Q_ASSERT( QApplication::instance()->thread() == QThread::currentThread() );
```

Comment:
can we leave the assertion here?

Reply:
if `QApplication::instance()->thread() != QThread::currentThread()` ->  `shared` is `false` (see just above), so this assertion can't be evaluate to `false`